test/pylib/scylla_cluster.py

```diff
@@ -165,9 +173,97 @@ def make_scylla_conf(mode: str, workdir: pathlib.Path, host_addr: str, seed_addr
     '--logger-log-level', 'raft_topology=debug',
     '--logger-log-level', 'query_processor=debug',
     '--logger-log-level', 'group0_raft_sm=trace',
-    '--logger-log-level', 'group0_voter_handler=debug',
 ]
 
+# A path to a Scylla executable, with version-specific
+# (i.e. not supported by *all* relevant versions)
+# Scylla config attached.
+class ScyllaVersionDescription(NamedTuple):
+    path: str # path to the Scylla executable
+    config: dict # a dictionary of added scylla.yaml options
+    argv: list[str] # a list of added CLI args
+
+# Returns the description of the current version.
+# (I.e. the one we just built and are now testing).
+#
+# (The path has to be passed by an argument because the current executable
+# has no fixed location -- the build directory depends on the build mode,
+# and in the case of cmake can be moved to something different than `build`.)
+def get_current_version_description(path: str) -> ScyllaVersionDescription:
+    return ScyllaVersionDescription(
+        path=path,
+        config={},
+        argv=[
+            '--logger-log-level', 'group0_voter_handler=debug',
+        ]
+    )
+
+@asynccontextmanager
+async def with_file_lock(lock_path: pathlib.Path) -> AsyncIterator[None]:
+    with open(lock_path, 'w') as f:
+        try:
+            await asyncio.to_thread(fcntl.flock, f, fcntl.LOCK_EX)
+            yield
+        finally:
+            await asyncio.to_thread(fcntl.flock, f, fcntl.LOCK_UN)
+
+async def get_scylla_2025_1_executable(build_mode: str) -> str:
+    async def run_process(cmd, **kwargs):
+        proc = await asyncio.create_subprocess_exec(*cmd, **kwargs)
+        await proc.communicate()
+        assert proc.returncode == 0
+
+    is_debug = build_mode == 'debug' or build_mode == 'sanitize'
+    package = "scylla-debug" if is_debug else "scylla"
+    arch = platform.machine()
+    url = f'https://downloads.scylladb.com/downloads/scylla/relocatable/scylladb-2025.1/{package}-2025.1.0-0.20250325.9dca28d2b818.{arch}.tar.gz'
```

Comment:
Are there are plans to extend version support?
I'm asking this because hostname and partial path can be extracted to the constant.

Reply:
> Are there are plans to extend version support?

Yes.

> I'm asking this because hostname and partial path can be extracted to the constant.

They can, but should they?

Reply:
Agree, we will change the path patterns from time to time, but won't invalidate old URLs.